File: JAVA Solutions/BinaryPrefixDivisibleByFive1018.java
import java.util.*;

public class BinaryPrefixDivisibleByFive1018 {
    public static void main(String[] args) {
        int[] nums = { 0, 1, 1, 1, 1, 1 }; // 0 2 4 8 16 32
        List<Boolean> res = prefixesDivBy5(nums);
        List<Boolean> ans = prefixesDivBySecond5(nums);
        System.out.println(res);
    }

    public static List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> res = new ArrayList<>();
        int num = nums[0];
        Boolean dBFive = num % 5 == 0 ? true : false;
        res.add(dBFive);
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == 1) {
                num = num * 2 + 1;
            } else {
                num = num * 2;
            }
            if (num % 5 == 0)
                res.add(true);
            if (num % 5 != 0)
                res.add(false);
            num = num % 5;
        }
        return res;
    }

    public static List<Boolean> prefixesDivBySecond5(int[] nums) {
        List<Boolean> ans = new ArrayList<>();
        int curr = 0;
        for (final int num : nums) {
            curr = (curr * 2 + num) % 5;
            ans.add(curr % 5 == 0);
        }
        return ans;
    }
}

File: C++ Solutions/BinaryPrefixDivisibleByFive1018.cpp
#include <bits/stdc++.h>
using namespace std;
vector<bool> prefixesDivBy5(vector<int> &nums)
{
    vector<bool> res;
    int num = nums[0];
    bool dBFive = num % 5 == 0 ? true : false;
    res.push_back(dBFive);
    for (int i = 1; i < nums.size(); i++)
    {
        if (nums[i] == 1)
        {
            num = num * 2 + 1;
        }
        else
        {
            num = num * 2;
        }
        if (num % 5 == 0)
            res.push_back(true);
        if (num % 5 != 0)
            res.push_back(false);
        num = num % 5;
    }
    return res;
}
vector<bool> prefixesDivBySecond5(vector<int>& nums) {
    vector<bool> ans;
    int curr = 0;

    for (const int num : nums) {
      curr = (curr * 2 + num) % 5;
      ans.push_back(curr % 5 == 0);
    }

    return ans;
  }
int main()
{
    vector<int> nums = {0, 1, 1, 1, 1, 1};
    vector<bool> res = prefixesDivBy5(nums);
    for (int i = 0; i < res.size(); i++)
    {
        cout << res[i] << " ";
    }
    return 0;
}